owid/datautils/format/numbers.py

```diff
@@ -0,0 +1,305 @@
+        """Build regex for a number with words.
+
+        Returns
+        -------
+        str
+            Regex.
+        """
+        regex = [
+            cls.regex_number_verbose_template.format(k, "|".join(v["words"]))
+            for k, v in cls.numeric_words.items()
+        ]
+        return r"\s?".join(regex)
+
+    @classmethod
+    def numeric_words_list(cls) -> Set[str]:
+        """Return list of all numeric words (flattened)."""
+        words = set(
+            word
+            for value in cls.numeric_words.values()
+            for word in value["words"]
+            if word != ""
+        )
+        return words
+
+    def _match_numeric_words(self) -> Dict[str, Union[str, int]]:
+        """Match number with words."""
+        match = re.search(self.regex_number_verbose(), self.number_raw)
+        if match:
+            numbers = match.groupdict(default=0)
+            return numbers
+        else:
+            raise ValueError("Number may not contain numeric words. Please review!")
+
+    @classmethod
+    def _build_number(cls, numbers: Dict[str, Union[str, int]]) -> int:
+        """Build number from dictionary."""
+        value = 0
+        for k, v in numbers.items():
+            value += float(v) * cls.numeric_words[k]["factor"]
+        return int(value)
+
+    @classmethod
+    def is_valid(cls, number: str) -> bool:
+        """Check if number contains numeric words.
+
+        Parameters
+        ----------
+        number : str
+            Candidate number.
+
+        Returns
+        -------
+        bool
+            True if number contains numeric valid words.
+        """
+        # return any(word in number for word in cls.numeric_words_list())
+        return bool(re.fullmatch(cls.regex_number_verbose(), number))
+
+    def clean(self) -> int:
+        """Clean number.
+
+        Returns:
+        --------
+        int:
+            Cleaned number.
+        """
+        if self.is_valid(self.number_raw):
+            number_dix = self._match_numeric_words()
+            number = self._build_number(number_dix)
+            return number
+        raise ValueError(f"Given number {self.number_raw} is not valid!")
+
+
+def num_to_str(num_as_str: Union[int, str]) -> str:
+    """Convert number to string.
+
+    Parameters
+    ----------
+    num_as_str: Union[int, str]
+        Raw number.
+
+    Returns:
+    --------
+    str:
+        Raw number as string.
+    """
+    if not isinstance(num_as_str, str):
+        return str(num_as_str)
+    return num_as_str
+
+
+def remove_multiple_whitespaces(text: str) -> str:
+    """Remove excess of whitespaces.
+
+    '  ' -> ' '.
+
+    Parameters
+    ----------
+    text : str
+        Raw text.
+
+    Returns
+    -------
+    str
+        String without duplicated whitespaces.
+    """
+    return re.sub(r"\s+", " ", text)
+
+
+def format_number(number: Union[int, str]) -> int:
+    """Format number.
+
+    Only supports integer conversion.
```

Comment:
Ah right, it's only meant for integers. If this is not expected to be generalised any time soon, maybe the name of the function should reflect it, e.g. `clean_integer`.

Reply:
I'll leave it as it is now, might add float support in the future.